test/error_in_creator.cpp

```diff
@@ -211,10 +212,10 @@ TEST_P(FaultyDelayedItemErrorTest, faultyCompressedItem)
   // (memory initialisation of the compression stream ?)
   usleep(1000000);
   // We detect it for any call after
-  EXPECT_THROW(creator.addMetadata("Title", "This is a title"), std::runtime_error);
-  EXPECT_THROW(creator.addIllustration(48, "PNGBinaryContent48"), std::runtime_error);
-  EXPECT_THROW(creator.addRedirection("foo2", "FooRedirection", "foo"), std::runtime_error);
-  EXPECT_THROW(creator.finishZimCreation(), std::runtime_error);
+  EXPECT_THROW(creator.addMetadata("Title", "This is a title"), AsyncError);
+  EXPECT_THROW(creator.addIllustration(48, "PNGBinaryContent48"), AsyncError);
+  EXPECT_THROW(creator.addRedirection("foo2", "FooRedirection", "foo"), AsyncError);
+  EXPECT_THROW(creator.finishZimCreation(), AsyncError);
```

Comment:
Shouldn't only the first one be `AsyncError`? After that the creator transitions to an error state and throws regular `std::runtime_error` exceptions on any attempt to use it. BTW, what about introducing a separate exception type for the latter case?

Reply:
Indeed. I've created a `CreatorStateError` for this.